src/cmakeTaskProvider.ts

```diff
@@ -189,15 +190,26 @@ export class CMakeTaskProvider implements vscode.TaskProvider {
     public static async findBuildTask(presetName?: string, targets?: string[]): Promise<CMakeTask | undefined> {
         // Fetch all CMake task from `tasks.json` files.
         const allTasks: vscode.Task[] = await vscode.tasks.fetchTasks({ type: CMakeTaskProvider.CMakeScriptType });
-        const tasks: (CMakeTask | undefined)[] = allTasks.map((task: any) => {
+
+        const project: CMakeProject | undefined = getActiveProject();
+        const cmakeDriver: CMakeDriver | undefined = (await project?.getCMakeDriverInstance()) || undefined;
+        const tasks: (CMakeTask | undefined)[] = await Promise.all(allTasks.map(async (task: any) => {
             if (!task.definition.label || !task.group || (task.group && task.group.id !== vscode.TaskGroup.Build.id)) {
                 return undefined;
             }
+
+            let taskTargets: string[];
+            if (cmakeDriver) {
+                taskTargets = await expand.expandStrings(task.definition.targets,  cmakeDriver.expansionOptions);
```

Comment:
the function findBuildTask is being called from the CMkae Driver, it's better if we pass the expansion options as the input to findBuildTask, instead of fetching the active project and getting the driver instance.

Reply:
Done.
FYI - i defined `expansionOptions` as optional parameter (and code checks if it was passed before calling expandString) - just in case when expansion is not required at all.